src/base.jl

```diff
@@ -7,11 +7,25 @@
 # -
 
 
+"""
+    AbstractIVP{T,Y}
+
+A progenitor of types representing an IVP (initial value
+problem).  The type parameters `T` and `Y` correspond to the types of
+time and state variable respectively.
+
+"""
 abstract AbstractIVP{T,Y}
-Base.eltype{T,Y}(::Type{AbstractIVP{T,Y}}) = T,Y,Y
+
+"""
+The "elements" of AbstractIVP are `t,y,dy` so the `eltype` returns `Tuple{T,Y,Y}`
+"""
+Base.eltype{T,Y}(::Type{AbstractIVP{T,Y}}) = Tuple{T,Y,Y}
 
 """
 
+    IVP{T,Y,F,G,J} <: AbstractIVP{T,Y}
```

Comment:
There is this: https://github.com/JuliaDocs/DocStringExtensions.jl

I think it can insert those signatures directly.